File: 2089-maximum-matrix-sum/maximum-matrix-sum.js
/**
 * @param {number[][]} matrix
 * @return {number}
 */
var maxMatrixSum = function (mat) {
    let n = mat.length;
    let tsum = 0, neg = 0, mn = Infinity;

    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (mat[i][j] < 0) {
                neg++;
            }

            tsum += Math.abs(mat[i][j]);
            mn = Math.min(mn, Math.abs(mat[i][j]));
        }
    }

    if (neg % 2 != 0) {
        return tsum -= 2 * mn;
    }

    return tsum;
};

File: 2089-maximum-matrix-sum/maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        tsum, neg, mn = 0, 0, float("inf")

        for row in matrix:
            for val in row:
                if val < 0:
                    neg += 1

                tsum += abs(val)
                mn = min(mn, abs(val))

        return (tsum - 2 * mn) if neg & 1 else tsum
